### ISSUE TITLE
Recurrences with a specific time can return recurrences in the past

### ISSUE THREAD

**Issue by natesalisbury** [rossta/montrose#110]
When passing the `at` option to a recurrence, the next recurrence returned will be for the current day even if the time has passed.

Example:

```ruby
r = Montrose.recurrence(every: :day, at: ["12pm", "4pm"])
=> #<Montrose::Recurrence:3ff9ef8196c8 {:every=>:day, :at=>[[12, 0, 0], [16, 0, 0]]}>
Time.now
=> 2019-03-14 16:24:05 -0600
r.take(3)
=> [2019-03-14 12:00:00 -0600, 2019-03-14 16:00:00 -0600, 2019-03-15 12:00:00 -0600]
```

Is this expected behavior? Is the design that it will return the first recurrence for the current day?

**Comment by rossta** [comment]
Yes, currently, the scope of the `every: :day` recurrence is set to match by date: 

https://github.com/rossta/montrose/blob/91af7d2d5c337aa255510179d24eb023775674bb/lib/montrose/frequency/daily.rb#L6-L8

... but I think it would make sense to change this to skip times occurring before the start time.

**Comment by rossta** [comment]
Fixed by #114
